src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -449,6 +467,50 @@ private EppoValue getSubjectVariationOverride(
         return experimentConfiguration.getTypedOverrides().getOrDefault(hexedSubjectKey, new EppoValue());
     }
 
+    /***
+     * Logs an action taken that was not selected by the bandit.
+     * Useful for full transparency on what users experienced.
+     * @param subjectKey subjectKey identifier of the experiment subject, for example a user ID.
+     * @param flagKey feature flag, bandit, or experiment identifier
+     * @param subjectAttributes optional attributes associated with the subject, for example name, email, account age, etc.
+     * @param actionString name of the action taken for the subject
+     * @param actionAttributes attributes associated with the given action
+     * @return null if no exception was encountered by logging; otherwise, the encountered exception
+     */
+    public Exception logNonBanditAction(
```

Comment:
For now, in lieu of a whole complex logging result object, I'll just return any exceptions encountered so customers can deal with them. I imagine in v3, this will return a logging result of sorts.